update and detail MV and DT support (#3779)
fishtown-analytics/docs.getdbt.com: website/docs/reference/resource-configs/postgres-configs.md

The Postgres adapter supports [materialized views](https://www.postgresql.org/docs/current/rules-materializedviews.html) and refreshes them for every subsequent `dbt run` you execute. For more information, see [Refresh Materialized Views](https://www.postgresql.org/docs/15/sql-refreshmaterializedview.html) in the Postgres docs.

Materialized views support the optional configuration `on_configuration_change` with the following values:

- `apply` (default) &mdash; attempts to update the existing database object if possible, avoiding a complete rebuild. The following index action can be applied without the need to rebuild the materialized view:

>     - Added

Reviewer: If we're using `dropped`, should this be `created`? I feel like `drop` and `create` go together, or `add` and `delete`. And then I'm on the fence about `updated`, which is just a `drop` followed by a `create`; you can't alter an index as far as I can tell, so `alter` would be misleading.

</File>

<File name='dbt_project.yml'>

```yaml
models:
  path:
    materialized: materialized_view
```

</File>

### Limitations

#### Changing materialization to and from "materialized_view"

> Swapping an already materialized model to be a materialized view and vise versa. The workaround is the manually drop the existing materialization in the data warehouse before calling `dbt run` again.

Reviewer: Is it worth calling out that `--full-refresh` doesn't resolve this issue either? I could see that being counterintuitive since `--full-refresh` feels to me like we're starting from a blank slate (but we're not).

Author: ```suggestion
Swapping an already materialized model to be a materialized view and vise versa. The workaround is the manually drop the existing materialization in the data warehouse before calling `dbt run` again. Normally, re-running with the `--full-refresh` flag would resolve this, but not in this case.
```

```
</File>

### Limitations

> 

Reviewer 2: We don't generally like to have two headers back-to-back. Is there a short sentence or two we can add here that would sprinkle in some commentary?

Author: how's this?

```suggestion

Below are current limitations that we hope to address in a future release.
```

```
</File>

### Limitations

Below are current limitations that we hope to address in a future release.
#### Changing materialization to and from "materialized_view"

> Swapping an already materialized model to be a materialized view and vise versa. The workaround is the manually drop the existing materialization in the data warehouse before calling `dbt run` again. Normally, re-running with the `--full-refresh` flag would resolve this, but not in this case.

Reviewer 2: ```suggestion
Swapping an already materialized model to a materialized view and vice versa. The workaround is manually dropping the existing materialization in the data warehouse before calling `dbt run` again. Normally, re-running with the `--full-refresh` flag would resolve this, but not in this case.
```

> To illustrate, assume for the example model below, `my_model`, that it has already been materialized to the underlying data platform via `dbt run`. If a user then changes the model's config to be `materialized="materialized_view"`, they will get an error. The workaround is to execute `DROP TABLE my_model` on the data warehouse before trying the model again.

Reviewer 2: ```suggestion
For example, assume the model below, `my_model`, has already been materialized to the underlying data platform via `dbt run`. If a user changes the model's config to `materialized="materialized_view"`, they will get an error. The solution is to execute `DROP TABLE my_model` on the data warehouse before trying the model again.
```
